lib/rubocop/cop/rails/action_name.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Use only specific action names.
+      #
+      # @example
+      #   # bad
+      #   class UsersController < ApplicationController
+      #     def articles
+      #     end
+      #   end
+      #
+      #   # good
+      #   class UserArticlesController < ApplicationController
+      #     def index
+      #     end
+      #   end
+      class ActionName < Base
```

Comment:
`Rails/RestfulActionName` might be more expressive.

Reply:
Hmm. I'd like to discuss the name some more. This is a topic that will be relevant to the detailed comments I will write in the Cop comments.

I am considering the following three or more candidates for the name of this Cop:

1. RestfulActionName
2. SpecificActionName
3. ActionName

The original background and motivation is that I want to use only the default action names for CRUD, but what this Cop is actually doing is just limiting the action names to specific predefined name set.

For example, if the convention emerged that there should be one controller class per endpoint, we could do that by limiting all action names to `#call`. Like this, the usage of this Cop is not necessarily limited to RESTful action names. This is why I gave it the rather generic name "ActionName."

If we believe that Cop name should be in keeping with the original good practice or idea, then "Restful" is a good name to use. Otherwise, if we are naming Cop with a focus on what it actually does, we would avoid "Restful" in this case.

If "ActionName" seems too generic, then I would suggest changing the name to something like "SpecificActionName" or "PredefinedActionName".

Reply:
I like the concept of this, but it's a tricky to decide what to call it.

I think we should having `Name` as part of the cop name. This isn't really about how you name the methods, it's about how you structure the app.

My suggestion is `Rails/ConstrictedActions`.

Reply:
The next release is a bugfix anyway. Let's take a moment to think of a more better name.

Reply:
Thank you for the comments about the cop naming.

I think `Rails/ActionName` is a bit too abstract and would be better to be at least more concrete. In that sense, I agree with the direction of `Rails/ConstrictedActions`. For now, I have chosen the Cop name `Rails/SpecificActionNames`, after the vocabulary of the offense message.

I don't think we should not force the RESTful concept too strongly on users, considering the users who use this simply to reduce the variety of method names.  It is difficult to nuance that, but it would be nice for the Cop to say something like "The default is RESTful oriented and we think it should be so if possible, but it is not mandatory".